backend/src/services/api-service/optimal/optimal-service.ts

```diff
@@ -28,6 +31,34 @@ export function findOptimalSetsForMeal(mealName: string, input: SetCoverProducti
   return customOptimalSet(mealName, input, TEAMFINDER_SET_COVER_TIMEOUT, monteCarloIterations);
 }
 
+/**
+ * Finds the optimal Pokemon for a specific ingredient
+ *
+ * API: /api/optimal/ingredient
+ */
+export function findOptimalMonsForIngredient(
+  ingredientName: string,
+  input: SetCoverProductionStats,
+  monteCarloIterations: number
+) {
+  const ingredient: Ingredient = getIngredientForName(ingredientName);
+  const pokemonProduction = calculateOptimalProductionForSetCover(input, monteCarloIterations);
+  const reverseIndex = createPokemonByIngredientReverseIndex(pokemonProduction);
+
+  const optimalCombinations = rankByIngredient({
+    ingredient,
+    reverseIndex,
+    cache: new Map(),
+    timeout: TEAMFINDER_SET_COVER_TIMEOUT,
+  });
+
+  return {
+    ingredient: ingredient.longName,
```

Comment:
In other parts of the site we currently use the short name more than the long name. You can check the current results on the site, they'll all list "potato" instead of "soft potato". 

I'll leave this decision to you though, but you're currently using the short name as input param in the route too like /api/optimal/ingredient/potato

Reply:
Changed to be consistent. In the future, I think "Fancy Apple Curry" and "Soft Potato" should be used in output rather than "FANCY_APPLE_CURRY" and "POTATO", but those changes can be made together, probably after the frontend redesign.